pkg/kubernetes/api/storage/v1/storageclass/build.go

```diff
@@ -186,6 +191,41 @@ func WithDevice() StorageClassOption {
 	}
 }
 
+func WithXfsQuota(softLimit, hardLimit string) StorageClassOption {
+	return func(s *storagev1.StorageClass) error {
+		if !isCompatibleWithXfsQuota(s) {
+			return errors.New("Failed to set XFSQuota parameters. " +
+				"Invalid existing '" + string(mconfig.CASConfigKey) + "' annotation" +
+				" parameters or Provisioner name.")
+		}
+
+		config := "- name: XFSQuota\n" +
```

Comment:
We can probably get back to this and refactor a bit. 

Can you add a todo comment here?

Reply:
Yep.